### ISSUE TITLE
Discrepancy between reported accuracy and predicted accuracy 

### ISSUE THREAD

**Issue by sfarkya** [antspy/quantized_distillation#18]
Hello, 

Thank you for sharing your codebase. 

So, I used this repo to train Resnet20 (All teacher, smaller, etc are all same) on cifar10 and I see some ambiguity in **reported performance vs predicted performance** in the last section of your evaluations here [line](https://github.com/antspy/quantized_distillation/blob/master/cifar10_wideResNet.py#L188).

**Here are the final results:** 
```
Model cifar10_distilled_quantized8bits prediected accuracy tensor(91.8600) reported accuract 91.91

Effective bit Huffman: 7.585104101179689 - Size reduction: 4.084183131042501 - Size MB : 0.26617525000000014

Model cifar10_teacher prediected accuracy tensor(90.5300) reported accuract 90.53
Size MB : 1.08164

Model cifar10_distilled_quantized4bits prediected accuracy tensor(91.4400) reported accuract 88.03

Effective bit Huffman: 3.542860840945231 - Size reduction: 8.436903261661763 - Size MB : 0.129758

Model cifar10_distilled_quantized2bits prediected accuracy tensor(90.6000) reported accuract 12.23

Effective bit Huffman: 1.5554269442698125 - Size reduction: 17.72433944312385 - Size MB : 0.06269675

Model cifar10_distilled_model prediected accuracy tensor(92.2500) reported accuract 92.25
Size MB : 1.08164

PM quantization of model "cifar10_distilled_model" with "2" bits and bucketing 256: 21.850000%

Effective bit Huffman: 1.4938500795088938 - Size reduction: 18.350201302288497 - Size MB: 0.06062125

PM quantization of model "cifar10_distilled_model" with "4" bits and bucketing 256: 89.990005%

Effective bit Huffman: 2.690225953182204 - Size reduction: 10.883517290692039 - Size MB: 0.10095149999999999

PM quantization of model "cifar10_distilled_model" with "8" bits and bucketing 256: 92.169998%

Effective bit Huffman: 6.505658074775342 - Size reduction: 4.7367702222057995 - Size MB: 0.22969087499999996

```
There's a huge difference especially in the case of **cifar10_distilled_quantized2bits**. It seems like some issues in evaluation. 
I see that pred_accuracy was hard-coded to 0. Is there any reason for that?

**Comment by antspy** [comment]
Hi sfarkya,

I don't remember why I had set pred_accuracy = 0, unfortunately I did not have time go go back and polish the code, so it has some weird leftovers from my experiments. 

If I remember correctly, the prediction accuracy vs reported accuracy issue comes from an inaccuracy in the training code, where I forget to quantize the model before evaluating during the train-test loop. This means that the numbers reported by the model manager are incorrect. Note that this only affects the numbers displayed during training and saved by the model manager; the training of the model itself is correct. However, this means that I have to recompute the model accuracy "live" and cannot rely on the one saved by the model manager.

The right number should be prediction accuracy, since the model is quantized before being saved by the model manager, hence you are evaluating a fully quantized model on the test set, which is what you want. About the discrepancy, I found that it is usually minor (as you see for 4 and 8 bits). I remember investigating why it was higher in some cases, but I don't recall my conclusions at the moment. In any case you can ignore it and focus on the evaluated accuracy.
